lib/Controller/SettingsController.php

```diff
@@ -47,20 +50,64 @@ class SettingsController extends Controller {
 	private $providerService;
 	/** @var ICrypto */
 	private $crypto;
+	/** @var IClientService */
+	private $clientService;
+	/** @var ILogger */
+	private $logger;
 
 	public function __construct(
 		IRequest $request,
 		ProviderMapper $providerMapper,
 		ID4MeService $id4meService,
 		ProviderService $providerService,
-		ICrypto $crypto
-		) {
+		ICrypto $crypto,
+		IClientService $clientService,
+		LoggerInterface $logger
+	) {
 		parent::__construct(Application::APP_ID, $request);
 
 		$this->providerMapper = $providerMapper;
 		$this->id4meService = $id4meService;
 		$this->providerService = $providerService;
 		$this->crypto = $crypto;
+		$this->clientService = $clientService;
+		$this->logger = $logger;
+	}
+
+	public function isDiscoveryEndpointValid($url) {
+		$result = [
+			'isReachable' => false,
+			'missingFields' => [],
+		];
+
+		try {
+			$client = $this->clientService->newClient();
+			$response = $client->get($url);
+			$httpCode = $response->getStatusCode();
+			$body = $response->getBody();
+
+			// Check if the request was successful
+			if ($httpCode == 200 && !empty($body)) {
```

Comment:
```suggestion
			if ($httpCode === Http::STATUS_OK && !empty($body)) {
```
Use triple equal and use `OCP\AppFramework\Http` constants for HTTP response codes.